crucible-mir/src/Mir/Trans.hs

```diff
@@ -2005,27 +2033,29 @@ genFn (M.Fn fname argvars _ body@(MirBody localvars blocks _)) rettype inputs =
 
   where
     initArgs :: HasCallStack =>
-        Ctx.Assignment (R.Atom s) args -> [M.Var] -> MirGenerator h s ret ()
+        [MirExp s] -> [M.Var] -> MirGenerator h s ret ()
     initArgs inputs vars =
-        case (Ctx.viewAssign inputs, vars) of
-            (Ctx.AssignEmpty, []) -> return ()
-            (Ctx.AssignExtend inputs' input, var : vars') -> do
+        case (inputs, vars) of
+            ([], []) -> return ()
+            (input : inputs', var : vars') -> do
                 mvi <- use $ varMap . at (var ^. varname)
                 Some vi <- case mvi of
                     Just x -> return x
                     Nothing -> mirFail $ "no varinfo for arg " ++ show (var ^. varname)
-                Refl <- testEqualityOrFail (R.typeOfAtom input) (varInfoRepr vi) $
+                MirExp inputTpr inputExpr <- return input
```

Comment:
Consider pattern-matching directly on `input` to avoid this `return input` statement:

```hs
            (MirExp inputTpr inputExpr : inputs', var : vars') -> do
```

Reply:
Fixed in 579afd3857198af9fc108870f387483beb58c677